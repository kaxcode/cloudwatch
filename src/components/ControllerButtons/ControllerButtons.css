.ControllerButtons {
  grid-area: controllerBtns;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  font-family: "Baloo Bhai", cursive;
}

.ControllerBtn__Start,
.ControllerBtn__Pause,
.ControllerBtn__Clear {
  margin-left: 30px;
  font-size: 3em;
  width: 290px;
  border-radius: 5px;
  padding: 20px;
  text-align: center;
  background-color: white;
  display: table-cell;
  vertical-align: middle;
  cursor: pointer;
}

.ControllerBtn__Start {
  border-style: solid;
  border-color: #41d2b9;
}

.ControllerBtn__Pause {
  border-style: solid;
  border-color: #f45b69;
}

.ControllerBtn__Clear {
  border-style: solid;
  border-color: #ffaa5a;
}
