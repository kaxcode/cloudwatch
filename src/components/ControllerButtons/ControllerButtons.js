import React from 'react';
import './ControllerButtons.css';
import { Button } from 'react-materialize';
import { func, bool } from 'prop-types';

const ControllerButtons = props => {
  return (
    <div className="ControllerButtons">
      <Button
        waves="light"
        className="ControllerBtn__Start green"
        id="timer-start"
        onClick={props.onStart}
      >
        Start
      </Button>

      <Button
        waves="light"
        className="ControllerBtn__Pause red"
        id="timer-pause"
        onClick={props.onPause}
      >
        Pause
      </Button>

      <Button
        waves="light"
        className="ControllerBtn__Clear yellow darken-3"
        id="timer-clear"
        onClick={props.onClear}
      >
        Clear
      </Button>
    </div>
  );
};

ControllerButtons.propTypes = {
  onStart: func.isRequired,
  onPause: func.isRequired,
  onClear: func.isRequired,
  show: bool
};

export default ControllerButtons;
