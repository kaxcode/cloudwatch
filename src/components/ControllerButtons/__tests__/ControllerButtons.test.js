import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ControllerButtons from '../ControllerButtons';

import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('ControllerButtons', () => {
  let wrapper;
  const onMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <ControllerButtons onPause={onMock} onStart={onMock} onClear={onMock} />
    );
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('the start button', () => {
    it('calls the :onStart prop when clicked', () => {
      //Arrange
      const startButton = wrapper.find('#timer-start').first();
      //Act
      startButton.simulate('click');
      //Assert
      expect(onMock).toHaveBeenCalled();
    });
  });

  describe('the pause button', () => {
    it('calls the :onPause prop when clicked', () => {
      //Arrange
      const pauseButton = wrapper.find('#timer-pause').first();
      //Act
      pauseButton.simulate('click');
      //Assert
      expect(onMock).toHaveBeenCalled();
    });
  });

  describe('the clear button', () => {
    it('calls the :onClear prop when clicked', () => {
      //Arrange
      const clearButton = wrapper.find('#timer-clear').first();
      //Act
      clearButton.simulate('click');
      //Assert
      expect(onMock).toHaveBeenCalled();
    });
  });
});
