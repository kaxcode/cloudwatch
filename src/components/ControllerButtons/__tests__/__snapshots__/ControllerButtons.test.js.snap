// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControllerButtons renders correctly 1`] = `
<div
  className="ControllerButtons"
>
  <button
    className="ControllerBtn__Start"
    id="timer-start"
    onClick={[Function]}
    waves="light"
  >
    Start
  </button>
  <button
    className="ControllerBtn__Pause"
    id="timer-pause"
    onClick={[Function]}
    waves="light"
  >
    Pause
  </button>
  <button
    className="ControllerBtn__Clear"
    id="timer-clear"
    onClick={[Function]}
    waves="light"
  >
    Clear
  </button>
</div>
`;
